class/defaults.yml

```diff
@@ -50,6 +50,9 @@ parameters:
         repository: kubebuilder/kube-rbac-proxy
         tag: v0.15.0
 
+    argocds: {}
+    projects: {}
```

Comment:
There's an argument that the config structure should be something like the following to more tightly couple appprojects with their argocd instance:

```yaml
    argocds: 
      some-namespace/some-argocd:
        config:
          spec: ...
        projects:
          some-project: ...
```

However, to do this, you'd have to do some extra work in the Jsonnet:

```jsonnet
local argocds = {
  [k]: std.get(params.argocds[k], 'config')
  for k in std.objectFields(params.argocds)
  if params.argocds[k] != null
};

local argocd_projects = {
  [k]: std.get(params.argocds[k], 'projects')
  for k in std.objectFields(params.argocds)
  if params.argocds[k] != null
};

local projects = {
  [namespacedName(k).namespace + '/' + pn]: argocd_projects[k][pn]
  for k in std.objectFields(argocd_projects)
  if argocd_projects[k] != null
  for pn in std.objectFields(argocd_projects[k])
  if argocd_projects[k][pn] != null
};

{
  '00_argocds': com.generateResources(argocds, ArgoCD), 
  '10_approjects': com.generateResources(projects, AppProject),
}
```

The big upside of this approach would be that you clearly define a link between ArgoCD and AppProject in the hierarchy.